### PR TITLE
feat: spring boot 3.3.1

### PR DESCRIPTION
**Pull request by burkedavison**
Spring Boot 3.3.x upgraded Brave and Zipkin dependencies by a major version with significant backward incompatibilities.  For this reason, the trace module which uses them is unable to be maintained with support for Spring Boot 3.2.x. 

Testing has been added to ensure Spring Boot 3.2.x support is maintained for all modules except Trace.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,10 +37,11 @@
 		<project.parent.version>${project.version}</project.parent.version>
 		<!-- Dependency versions -->
```
Comment (burkedavison) [inline pom.xml:38]: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/9651092003/job/26654930114?pr=2990
```
Caused by: com.google.cloud.sql.core.TerminalException: [spring-cloud-gcp-ci-native:us-central1:testmysql] The Google Cloud SQL Admin API failed for the project "spring-cloud-gcp-ci-native". Reason: 403 Forbidden
GET https://sqladmin.googleapis.com/sql/v1beta4/projects/spring-cloud-gcp-ci-native/instances/testmysql/connectSettings
***
  "code": 403,
  "errors": [
    ***
      "domain": "global",
      "message": "boss::NOT_AUTHORIZED: Not authorized to access resource. Possibly missing permission cloudsql.instances.get on resource instances/testmysql.",
      "reason": "forbidden"
    ***
  ],
  "message": "boss::NOT_AUTHORIZED: Not authorized to access resource. Possibly missing permission cloudsql.instances.get on resource instances/testmysql.",
  "status": "PERMISSION_DENIED"
***
	at com.google.cloud.sql.core.DefaultConnectionInfoRepository.addExceptionContext(DefaultConnectionInfoRepository.java:456)
	at com.google.cloud.sql.core.DefaultConnectionInfoRepository.fetchMetadata(DefaultConnectionInfoRepository.java:297)
	at com.google.cloud.sql.core.DefaultConnectionInfoRepository.lambda$getConnectionInfo$0(DefaultConnectionInfoRepository.java:135)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

File: pom.xml [hunk 2]
```diff
@@ -37,10 +37,11 @@
...
 		<spring-cloud-dependencies.version>2023.0.2</spring-cloud-dependencies.version>
```
Comment (burkedavison) [inline pom.xml:39]: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/9651092003/job/26654930547?pr=2990
```
Error: Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///home/runner/.m2/repository/com/google/cloud/sql/jdbc-socket-factory-core/1.19.0/jdbc-socket-factory-core-1.19.0.jar!/META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/native-image.properties'. Use -H:PrintFlags= to list all available options.
Error: Image build request failed with exit status 1
13:25:46:419 [INFO] ------------------------------------------------------------------------
13:25:46:419 [INFO] Reactor Summary for Spring Framework on Google Cloud Code Sample - Postgres 5.4.2-SNAPSHOT:
13:25:46:419 [INFO] 
13:25:46:419 [INFO] Spring Framework on Google Cloud Code Sample - Postgres FAILURE [ 13.090 s]
```

Comment (burkedavison) [reply pom.xml:39]: https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/releases/tag/v1.19.0

Release notes:
> Update native-image.version to v0.10.2 ([dbffbf4](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/commit/dbffbf48c38b25f47b85fa50e4b938ebf88f9357))

Comment (burkedavison) [reply pom.xml:39]: `UnlockExperimentalVMOptions` enabled here:
https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/38b2b1825df3e43ed5775c8cc894aaa037f6a347/core/src/main/resources/META-INF/native-image/com.google.cloud.sql/cloud-sql-jdbc-socket-factory-parent/native-image.properties#L2

Comment (burkedavison) [reply pom.xml:39]: Native tests skipped in release PR and main branch:
image
image

File: pom.xml [hunk 3]
```diff
@@ -37,10 +37,11 @@
 		<project.parent.version>${project.version}</project.parent.version>
```
[the remark this answers is not in the document]
Comment (burkedavison) [reply pom.xml:37]: Resolved by rotating keys

File: pom.xml (hunk 1 above)
Comment (burkedavison) [reply pom.xml:38]: resolved by rotating keys

File: pom.xml (hunk 2 above)
Comment (burkedavison) [reply pom.xml:39]: Introduced in https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/pull/1961

Comment (burkedavison) [reply pom.xml:39]: Flag was introduced into GraalVM CE @ 23.1.1: https://github.com/oracle/graal/blob/vm-ce-23.1.1/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java#L1015-L1016

Spring Cloud GCP tests with GraalVM 22.3.2: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/.github/workflows/NativeTests.yaml#L84

Comment (burkedavison) [reply pom.xml:39]: According to https://github.com/graalvm/container/pkgs/container/graalvm-ce, the latest available docker image for graalvm-ce is 22.3.3.

GraalVM community switches to JDK-based versioning.

Comment (burkedavison) [reply pom.xml:39]: cloud-sql-jdbc-socket-factory is performing scheduled native tests against graalvm-jdk-17.0.11+7.1

https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/actions/runs/9655336140/job/26631058934

**Comment by breun** [comment]
If Spring Cloud GCP 5.x aims to maintain compatibility with Spring Boot 3.2.x and 3.3.x, it could be dangerous to upgrade to Spring Boot 3.3.x. Sticking with 3.2.x would avoid accidentally using API which is not available in Spring Boot 3.2.x.

Or will compatibility with Spring Boot 3.2.x be guaranteed in some other way?

**Comment by burkedavison** [comment]
@breun : Agreed with your concern. We'll have an answer before merging.

**Comment by burkedavison** [comment]
image

Compatibility table updates from https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/3000 merged here + added an exception for trace module compatibility. Main branch testing now include Spring Boot 3.2 sample tests.

**Comment by burkedavison** [comment]
Fixes #2917

**Pull request merged**